examples/js/effects/OutlineEffect.js

```diff
@@ -0,0 +1,294 @@
+
+		var materials = [];
+
+		for ( var i = 0, il = originalMaterial.materials.length; i < il; i ++ ) {
+
+			materials.push( createMaterial( originalMaterial.materials[ i ] ) );
+
+		}
+
+		return new THREE.MultiMaterial( materials );
+
+	}
+
+	function setOutlineMaterial ( object ) {
+
+		if ( object.material === undefined ) return;
+
+		object.userData.originalMaterial = object.material;
+
+		if ( object.userData.outlineMaterial === undefined ) {
+
+			object.userData.outlineMaterial = object.material.type === 'MultiMaterial' ? createMultiMaterial( object.material ) : createMaterial( object.material );
+
+		}
+
+		if ( object.userData.outlineMaterial.type === 'MultiMaterial' ) {
+
+			updateOutlineMultiMaterial( object.userData.outlineMaterial, object.userData.originalMaterial );
+
+		} else {
+
+			updateOutlineMaterial( object.userData.outlineMaterial, object.userData.originalMaterial );
+
+		}
+
+		object.material = object.userData.outlineMaterial;
+
+	}
+
+	function updateOutlineMaterial ( material, originalMaterial ) {
+
+		if ( material === invisibleMaterial ) return;
+
+		var outlineParameters = originalMaterial.outlineParameters;
+
+		material.skinning = originalMaterial.skinning;
+		material.morphTargets = originalMaterial.morphTargets;
+		material.morphNormals = originalMaterial.morphNormals;
+		material.fog = originalMaterial.fog;
+		material.visible = originalMaterial.visible;
+		material.uniforms.outlineAlpha.value = originalMaterial.opacity;
+
+		if ( outlineParameters !== undefined ) {
+
+			if ( outlineParameters.thickness !== undefined ) material.uniforms.outlineThickness.value = outlineParameters.thickness;
+			if ( outlineParameters.color !== undefined ) material.uniforms.outlineColor.value.copy( outlineParameters.color );
+			if ( outlineParameters.alpha !== undefined ) material.uniforms.outlineAlpha.value = outlineParameters.alpha;
+			if ( outlineParameters.visible !== undefined ) material.visible = outlineParameters.visible;
+
+		}
+
+		if ( material.uniforms.outlineAlpha.value < 1.0 ) material.transparent = true;
+
+	}
+
+	function updateOutlineMultiMaterial ( material, originalMaterial ) {
+
+		var outlineParameters = originalMaterial.outlineParameters;
+
+		material.visible = originalMaterial.visible;
+
+		if ( outlineParameters !== undefined ) {
+
+			if ( outlineParameters.visible !== undefined ) material.visible = outlineParameters.visible;
+
+		}
+
+		for ( var i = 0, il = material.materials.length; i < il; i ++ ) {
+
+			updateOutlineMaterial( material.materials[ i ], originalMaterial.materials[ i ] );
+
+		}
+
+	}
+
+	function restoreOriginalMaterial ( object ) {
+
+		if ( object.userData.originalMaterial !== undefined ) object.material = object.userData.originalMaterial;
+
+	}
+
+	this.setSize = function ( width, height ) {
+
+		renderer.setSize( width, height );
+
+	};
+
+	this.render = function ( scene, camera, renderTarget, forceClear ) {
+
+		var tmpAutoClear = renderer.autoClear;
```

Comment:
Minor thing but... we usually call this `var currentAutoClear = renderer.autoClear;`.